third_party/move/move-vm/runtime/src/native_functions.rs

```diff
@@ -176,19 +188,40 @@ impl<'a, 'b> NativeContext<'a, 'b> {
     }
 
     pub fn type_to_type_layout(&self, ty: &Type) -> PartialVMResult<MoveTypeLayout> {
-        StorageLayoutConverter::new(self.module_storage).type_to_type_layout(ty)
+        Ok(self
+            .type_to_type_layout_with_delayed_field_check(ty)?
+            .unpack()
+            .0)
     }
 
-    pub fn type_to_type_layout_with_identifier_mappings(
+    pub fn type_to_type_layout_with_delayed_field_check(
         &self,
         ty: &Type,
-    ) -> PartialVMResult<(MoveTypeLayout, bool)> {
-        StorageLayoutConverter::new(self.module_storage)
-            .type_to_type_layout_with_identifier_mappings(ty)
+    ) -> PartialVMResult<LayoutWithDelayedFields> {
+        let loader = NativeLoader::new(self.module_storage);
+        LayoutConverter::new(&loader).type_to_type_layout_with_delayed_field_check(
+            // Gas meter will be ignored here. Passing as a no-op. All modules are expected to
+            // be visited.
+            &mut UnmeteredGasMeter,
+            &mut UnvisitableModuleTraversalContext::new(self.traversal_context),
+            ty,
+        )
     }
 
-    pub fn type_to_fully_annotated_layout(&self, ty: &Type) -> PartialVMResult<MoveTypeLayout> {
-        StorageLayoutConverter::new(self.module_storage).type_to_fully_annotated_layout(ty)
+    pub fn type_to_fully_annotated_layout(
+        &self,
+        ty: &Type,
+    ) -> PartialVMResult<Option<MoveTypeLayout>> {
+        let loader = NativeLoader::new(self.module_storage);
```

Comment:
wouldn't it be cleaner to call the above method then do into_layout_expect_no_delayed_field on the result?

Reply:
What do you mean by above? This one is annotated, and the other one is not?

Reply:
I mean type_to_type_layout_with_delayed_field_check
I might be missing smt but it seems like type_to_fully_annotated_layout could just be
`self.type_to_type_layout_with_delayed_field_check.map(into_layout_expect_no_delayed_fields)`
?

Reply:
But these are different? One is type_to_type_layout_with_delayed_field_check and the other one is type_to_**annotated**_type_layout_with_delayed_field_check? So the code is only duplicated because of annotated part which sets the const bool parameter differently to add field names to layout. I can propagate the const bool up the stack, maybe?